dNet/ClientPackets.cpp

```diff
@@ -284,17 +287,77 @@ void ClientPackets::HandleChatModerationRequest(const SystemAddress& sysAddr, Pa
 		receiver.push_back(static_cast<uint8_t>(character));
 	}
 
+	if (!receiver.empty()) {
+		if (std::string(receiver.c_str(), 4) == "[GM]") {
+			receiver = std::string(receiver.c_str() + 4, receiver.size() - 4);
+		}
+	}
+
 	stream.Read(messageLength);
 	for (uint32_t i = 0; i < messageLength; ++i) {
 		uint16_t character;
 		stream.Read(character);
 		message.push_back(static_cast<uint8_t>(character));
 	}
 
+	bool isBestFriend = false;
+
+	if (chatLevel == 1) {
+		// Private chat
+		LWOOBJID idOfReceiver = LWOOBJID_EMPTY;
+
+		{
+			sql::PreparedStatement* stmt = Database::CreatePreppedStmt("SELECT name FROM charinfo WHERE name = ?");
```

Comment:
I recommend changing this to be a unique_ptr so when you leave scope this memory is automatically managed.  Right now its a memory leak since it is not deleted.

Reply:
I don't really like using unique pointers, I have added the deletes back.